src/threading.ts

```diff
@@ -0,0 +1,110 @@
+import interfaces = require('./interfaces');
+import IPy_Object = interfaces.IPy_Object;
+import Py_CodeObject = require('./codeobject');
+import Py_FuncObject = require('./funcobject');
+import opcodes = require('./opcodes');
+import optable = require('./optable');
+import Py_Cell = require('./cell');
+import Py_FrameObject = require('./frameobject');
+import enums = require('./enums');
+
+var maxMethodResumes: number = 10000,
+// The number of method resumes until Doppio should yield again.
+methodResumesLeft: number = maxMethodResumes,
+// How responsive Ninia should aim to be, in milliseconds.
+responsiveness: number = 1000,
+// Used for the CMA.
+numSamples: number = 1;
+
+class Thread{
+	// Current state of Thread
+	private status: enums.ThreadStatus = enums.ThreadStatus.RUNNING;
```

Comment:
While this isn't a problem now with a single thread, newly created threads should be in the `NEW` state. They transition to `RUNNABLE` once they have a method to run & are commanded to run.

Think of how threads are created in e.g. Java:

```{java}
public class HelloThread extends Thread {

    public void run() {
        // ht is RUNNING
        System.out.println("Hello from a thread!");
        // once run returns, ht is TERMINATED
    }

    public static void main(String args[]) {
        // Create a new thread.
        var ht = new HelloThread();
        // ht is now in the NEW state. It is not running.
        // tell it to run:
        ht.run();
        // ht is now RUNNABLE.
    }

}
```

It's quite similar in Python

```{python}
import threading

def worker():
    """thread worker function"""
    print 'Worker'
    return

t = threading.Thread(target=worker)
t.start()
```

It might be best to modify the thread so it behaves like that, even if it doesn't matter right this moment.